frontend/src/components/Order/Order.jsx

```diff
@@ -1,231 +1,65 @@
 import React, { useState } from 'react';
-import { useForm } from 'react-hook-form';
 import styles from './Order.module.scss';
-import Button from '../ui-kit/components/Button';
-import { ICONS } from '../ui-kit/icons';
-import { IMaskInput } from 'react-imask';
 import OrderInfo from './components/OrderInfo';
+import DeliverForm from './components/DeliveryForm/DeliveryForm';
+import OrderSection from './components/OrderSection';
+import ContactDetails from './components/ContactDetails/ContactDetails';
 
 const Order = () => {
-  const [showForm, setShowForm] = useState(false);
-  const [hideForm, setHideForm] = useState(false);
-  const [client, setClient] = useState('');
-  const [phone, setPhone] = useState('');
-  const {
-    register,
-    formState: { errors },
-    setValue,
-    getValues,
-    handleSubmit,
-  } = useForm({
-    defaultValues: { recipient: 'I' },
-    mode: 'onBlur',
-  });
+  const [currentStep, setCurrentStep] = useState(1);
+  const [orderData, setOrderData] = useState({});
 
-  const onSubmitButton = (data) => {
-    setClient(data.clientFirstName);
-    setPhone(data.clientPhone);
-    setHideForm(true);
+  const handleChangeStep = (sourceStep) => {
+    setCurrentStep(sourceStep);
   };
 
-  const handleChangeData = () => {
-    setHideForm(false);
+  const handleStepFinish = (stepDone) => {
+    stepDone++;
+    setCurrentStep(stepDone);
   };
 
   return (
     <div className={styles.order}>
       <h2 className={styles.title}>Оформлення замовлення</h2>
       <div className={styles.mainGrid}>
-        <div className={styles.contactDetails}>
-          <div className={styles.subtitleBlock}>
-            <h3 className={styles.subtitle}>
-              <span>1.</span>Контактні дані
-            </h3>
-            {hideForm && (
-              <div className={styles.buttonChange}>
-                <Button
-                  variant="no-border"
-                  label="Змінити"
-                  icon={<ICONS.PencilIcon />}
-                  onClick={() => handleChangeData()}
-                ></Button>
-              </div>
-            )}
+        <div>
+          <div className={styles.orderBlock}>
+            <OrderSection
+              step={1}
+              currentStep={currentStep}
+              name={'Контактні дані'}
+              outputString={orderData.contactData?.outputString}
+              handleBackEdit={handleChangeStep}
+            >
+              <ContactDetails
+                contactData={orderData.contactData}
+                setDataOnSubmit={(newContactData) => {
+                  handleStepFinish(currentStep);
```

Comment:
You can create separate function and pass there and object with data that need to be added. You can use this function here and for delivery details also.

Reply:
Here new fields are added to the object, so it's more convenient (clearer) not to put this part in a separate function